### PR TITLE
Adding support for shared test containers

### PR DESCRIPTION
**Pull request by IsaacCalligeros95**
This change aims to support running the terraform provider tests against a single container. We're migrating the terraform provider and there's a lot of tests at the provider level that currently do not run in CI and expect a local instance. This is handled in the provider with [this change](https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/pull/665) namely the `testing_container_test.go` file (the name needs some thinking...). This should just work with the Ci parallel runs. 

To do this ArrangeTest has been copied and split out into `ArrangeContainer` and `CleanUp`. This is still a WIP, with a couple of edge cases (I think there's an issue with stopping the logger properly) and some need for clean-up (I've just copied across and duplicated a fair bit of the existing code). 

**TLDR:** Any concerns using shared containers and where required splitting tests by spaces? This can be supported in parallel to the existing per test container approach.

### CODE REVIEW

**Comment by IsaacCalligeros95** [comment]
@mcasperson If you have the time would you mind giving some high level feedback on this change? We're looking at adding the existing provider level tests + migration tests and would like to run these against a single instance for quicker feedback. If you have any major concerns I can close this.

File: test/octopus_container_test_framework.go [hunk 1]
```diff
@@ -43,7 +44,7 @@ type OctopusContainer struct {
 	URI string
 }
 
-type mysqlContainer struct {
+type MysqlContainer struct {
```
Comment (benPearce1) [inline test/octopus_container_test_framework.go:47]: mysql ?

**Review by HuyPhanNguyen — APPROVED** [review]
LGTM

File: test/octopus_container_test_framework.go [hunk 2]
```diff
@@ -283,6 +284,121 @@ func (o *OctopusContainerTest) createDockerInfrastructure(t *testing.T, ctx cont
...
+	if sqlTerminateErr := sqlServer.Container.Terminate(ctx); sqlTerminateErr != nil {
+		log.Printf("Failed to terminate the SQL Server container: %v", sqlTerminateErr)
+	}
+
+	if networkErr := network.Remove(ctx); networkErr != nil {
+		log.Printf("Failed to remove network: %v", networkErr)
+		return networkErr
+	}
+
+	return nil
```
Comment (HuyPhanNguyen) [inline test/octopus_container_test_framework.go:398]: Why only return networkErr and other error return nil?

File: test/octopus_container_test_framework.go (hunk 1 above)
Comment (IsaacCalligeros95) [reply test/octopus_container_test_framework.go:47]: I made that public to return it from the arrange method for cleanup. It's unlikely to be used but I figured I was better off exposing all of the containers/state in the method and cleaning up based off of that rather than some global state.

File: test/octopus_container_test_framework.go (hunk 2 above)
Comment (IsaacCalligeros95) [reply test/octopus_container_test_framework.go:398]: I've just removed the error return.

**Review by benPearce1 — APPROVED** [review]
LGTM

File: test/octopus_container_test_framework.go (hunk 1 above)
Comment (benPearce1) [reply test/octopus_container_test_framework.go:47]: I was actualy referring to the name

**Comment by IsaacCalligeros95** [comment]
We've decided to merge this to keep moving forward with the terraform provider changes. If there are any concerns or issues please let us know. There will likely be some additional minor changes in the near future.
